src/prefect_server/api/states.py

```diff
@@ -129,6 +130,10 @@ async def set_flow_run_state(
     if state.is_running() or state.is_submitted():
         await api.runs.update_flow_run_heartbeat(flow_run_id=flow_run_id)
 
+    # Set agent ID on flow run when submitted by agent
+    if state.is_submitted() and agent_id:
+        await api.runs.update_flow_run_agent(flow_run_id=flow_run_id, agent_id=agent_id)
```

Comment:
If the agent ID is invalid for any reason (including being valid and then having the agent deleted from the db), it will error and abort the state setting. Perhaps `update_flow_run_agent` should return `False` on error, instead of raising ValueError?

Reply:
Updated but FWIW the heartbeat update functions the same way. Not sure if that should change also